scale/source/models.py

```diff
@@ -139,6 +140,18 @@ def get_source_products(self, source_file_id, started=None, ended=None, batch_id
         :type is_superseded: bool
         :param file_name: Query product files with the given file name.
         :type file_name: str
+        :keyword job_output: Query product files with the given job output
+        :type job_output: str
+        :keyword recipe_ids: Query product files produced by a given recipe id
+        :type recipe_ids: list[int]
+        :keyword recipe_job: Query product files produced by a given recipe name
+        :type recipe_job: str
+        :keyword recipe_type: Query product files produced by a given recipe type
+        :type recipe_type: str
+        :keyword source_started: Query product files by a given start time for all source files
+        :type source_started: datetime str
+        :keyword source_ended: Query product files by a given stop time for all source files
+        :type source_ended: datetime str
         :param order: A list of fields to control the sort order.
```

Comment:
The new parameters need to be passed to filter_products(). Also remove source_started and source_ended and use time_field instead like filter_products().